uvicorn/config.py

```diff
@@ -366,37 +366,75 @@ def setup_event_loop(self):
-        addr_format = "%s://%s:%d"
+        if self.uds:
+            path = self.uds
+            sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            try:
+                sock.bind(path)
+                uds_perms = 0o666
+                os.chmod(self.uds, uds_perms)
+            except OSError as exc:
+                logger.error(exc)
+                sys.exit(1)
 
-        if self.host and ":" in self.host:
-            # It's an IPv6 address.
-            family = socket.AF_INET6
-            addr_format = "%s://[%s]:%d"
+            message = "Uvicorn running on unix socket %s (Press CTRL+C to quit)"
+            sock_name_format = "%s"
+            color_message = (
+                "Uvicorn running on "
+                + click.style(sock_name_format, bold=True)
+                + " (Press CTRL+C to quit)"
+            )
+            logger.info(
+                message,
+                self.uds,
+                extra={"color_message": color_message},
+            )
+        elif self.fd:
+            sock = socket.fromfd(self.fd, socket.AF_UNIX, socket.SOCK_STREAM)
+            message = "Uvicorn running on socket %s (Press CTRL+C to quit)"
+            fd_name_format = "%s"
+            color_message = (
+                "Uvicorn running on "
+                + click.style(fd_name_format, bold=True)
+                + " (Press CTRL+C to quit)"
+            )
+            logger.info(
+                message,
+                sock.getsockname(),
+                extra={"color_message": color_message},
+            )
+        else:
+            family = socket.AF_INET
+            addr_format = "%s://%s:%d"
+
+            if self.host and ":" in self.host:
+                # It's an IPv6 address.
+                family = socket.AF_INET6
+                addr_format = "%s://[%s]:%d"
+
+            sock = socket.socket(family=family)
+            sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            try:
+                sock.bind((self.host, self.port))
+            except OSError as exc:
+                logger.error(exc)
+                sys.exit(1)
 
-        sock = socket.socket(family=family)
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        try:
-            sock.bind((self.host, self.port))
-        except OSError as exc:
-            logger.error(exc)
-            sys.exit(1)
+            message = f"Uvicorn running on {addr_format} (Press CTRL+C to quit)"
+            color_message = (
+                "Uvicorn running on "
+                + click.style(addr_format, bold=True)
+                + " (Press CTRL+C to quit)"
+            )
+            protocol_name = "https" if self.is_ssl else "http"
+            logger.info(
+                message,
+                protocol_name,
+                self.host,
+                self.port,
+                extra={"color_message": color_message},
+            )
         sock.set_inheritable(True)
```

Comment:
Does this impact the socket from `socket.fromfd`? https://docs.python.org/3/library/socket.html#socket.fromfd

![image](https://user-images.githubusercontent.com/7353520/121392324-8f608980-c94f-11eb-988c-d92efc5395c3.png)

Reply:
this `sock.set_inheritable(True)` applies to all 3 cases so yes. it serves when we have multiple process and 1 socket to listen on iirc

Reply:
Ah, I understand what confused me now. Thanks.